Helper/PaymentResponseHandler.php

```diff
@@ -207,8 +207,10 @@ public function handlePaymentsDetailsResponse(
              * if no additional action is required according to /paymentsDetails response.
              * Otherwise keep the order state as pending_payment.
              */
-            $order = $this->orderHelper->setStatusOrderCreation($order);
-            $this->orderRepository->save($order);
+            if (!$order->isCanceled()) {
```

Comment:
Hello @dimitriBouteille,

Thank you for your contribution. Obviously, this fix solves the issue for order cancelled cases. However, I would like to recommend a holistic approach as we also have some reports for other statuses rather than only `cancelled`.

So, instead of checking whether if the order is already cancelled in the inner if block, I would recommend removing the inner if block and adding a second condition to the if block on line 204 to check if the order state is `pending_payment` or not. Because, the logic in the outer if block should only run under the condition of order should be at `pending_payment` state AND order's `resultCode` should not be in `ACTION_REQUIRED_STATUSES`.

Best regards,
Can

Reply:
Hello @dimitriBouteille,

Thank you for your contribution. That looks great and solves the issue indeed. After the reviews, this PR will be merged and released.

Best Regards,
Can

Reply:
Perfect :rocket: Is it possible to include it in the next release ([Release v9.7.3](https://github.com/Adyen/adyen-magento2/pull/2722)) ?

Reply:
Hello @dimitriBouteille! 

Yes, we will include the PR in the coming release.

Best Regards,
Can